### PR TITLE
Woe And Suffering/Atomization - Ports some tgstation controller code

### PR DESCRIPTION
**Pull request by mc-oofert**
## About The Pull Request

**TESTMERGE FIRST YA FUCKWADS**

holy shit how much i hate this
anyway atomized pr 
The following controllers were replaced with tgstation ones with alterations to work with TGMC: Admin, Failsafe, Master, Globals
Most if not all subsystems have been changed to support the new code

holy shit is there a lot to link

## Why It's Good For The Game

new code good + atomized pr + pilltadpole needs this to work + ratio 

## Changelog
should be none but there may so testmerge

### CODE REVIEW

File: code/__DEFINES/MC.dm [hunk 1]
```diff
@@ -72,11 +99,37 @@
...
+/datum/controller/subsystem/processing/##X/fire() {..() /*just so it shows up on the profiler*/} \
 /datum/controller/subsystem/processing/##X
+
+#define FLUID_SUBSYSTEM_DEF(X) GLOBAL_REAL(SS##X, /datum/controller/subsystem/fluids/##X);\
+/datum/controller/subsystem/fluids/##X/New(){\
+	NEW_SS_GLOBAL(SS##X);\
+	PreInit();\
+}\
+/datum/controller/subsystem/fluids/##X/fire() {..() /*just so it shows up on the profiler*/} \
+/datum/controller/subsystem/fluids/##X
```
Comment (ivanmixo) [inline code/__DEFINES/MC.dm:127]: We have a fluids subsystem?

File: code/__DEFINES/subsystems.dm [hunk 2]
```diff
@@ -0,0 +1,309 @@
...
+
+//! ## DB defines
+/**
+ * DB major schema version
+ *
+ * Update this whenever the db schema changes
+ *
+ * make sure you add an update to the schema_version stable in the db changelog
+ */
+#define DB_MAJOR_VERSION 5
```
Comment (ivanmixo) [inline code/__DEFINES/subsystems.dm:14]: This needs to be matched up with current db

File: code/__DEFINES/MC.dm [hunk 3]
```diff
@@ -72,11 +99,37 @@
 	NEW_SS_GLOBAL(SS##X);\
 	PreInit();\
 }\
+/datum/controller/subsystem/timer/##X/fire() {..() /*just so it shows up on the profiler*/} \
 /datum/controller/subsystem/timer/##X
 
+#define MOVEMENT_SUBSYSTEM_DEF(X) GLOBAL_REAL(SS##X, /datum/controller/subsystem/movement/##X);\
```
Comment (ivanmixo) [inline code/__DEFINES/MC.dm:105]: We have a movement subsystem?

File: code/__DEFINES/subsystems.dm [hunk 4]
```diff
@@ -0,0 +1,309 @@
...
+#define WARDROBE_CACHE_CALL_REMOVAL 4
+
+// Wardrobe preloaded stock indexes
+#define WARDROBE_STOCK_CONTENTS 1
+#define WARDROBE_STOCK_CALL_INSERT 2
+#define WARDROBE_STOCK_CALL_REMOVAL 3
+
+// Wardrobe callback master list indexes
+#define WARDROBE_CALLBACK_INSERT 1
+#define WARDROBE_CALLBACK_REMOVE 2
```
Comment (ivanmixo) [inline code/__DEFINES/subsystems.dm:274]: I don't think we have a wardrobe SS so these are just pointless.

File: code/__DEFINES/subsystems.dm [hunk 5]
```diff
@@ -0,0 +1,309 @@
...
+
+// Subsystem delta times or tickrates, in seconds. I.e, how many seconds in between each process() call for objects being processed by that subsystem.
+// Only use these defines if you want to access some other objects processing delta_time, otherwise use the delta_time that is sent as a parameter to process()
+#define SSFLUIDS_DT (SSplumbing.wait/10)
```
Comment (ivanmixo) [inline code/__DEFINES/subsystems.dm:278]: Again fluid subsystem.

File: code/__DEFINES/subsystems.dm [hunk 6]
```diff
@@ -0,0 +1,309 @@
...
+#define SSAIR_ACTIVETURFS 3
+#define SSAIR_HOTSPOTS 4
+#define SSAIR_EXCITEDGROUPS 5
+#define SSAIR_HIGHPRESSURE 6
+#define SSAIR_SUPERCONDUCTIVITY 7
+#define SSAIR_PROCESS_ATOMS 8
+
+// Pipeline rebuild helper defines, these suck but it'll do for now //Fools you actually merged it
+#define SSAIR_REBUILD_PIPELINE 1
+#define SSAIR_REBUILD_QUEUE 2
```
Comment (ivanmixo) [inline code/__DEFINES/subsystems.dm:250]: We don't have atmos , make sure we actually need these

**Review by TiviPlus — CHANGES_REQUESTED** [review]
This is unreviewable you need to split these and link prs if you are porting

**Comment by mc-oofert** [comment]
> This is unreviewable you need to split these and link prs if you are porting

skill issue

**Pull request closed without merging**